### ISSUE TITLE
VerifyError: Cannot inherit from final class

### ISSUE THREAD

**Issue by bartoszwilk** [andrzejchm/RESTMock#56]
**Kotlin version**: 1.1.0
**RESTMock Version**: 0.2.1

Hi

I'm trying to use your library in Kotlin with JUnit robolectric tests and getting following exception:

```ruby
Exception in thread "pool-3-thread-1" java.lang.VerifyError: Cannot inherit from final class
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
	at org.robolectric.internal.bytecode.SandboxClassLoader.maybeInstrumentClass(SandboxClassLoader.java:138)
	at org.robolectric.internal.bytecode.SandboxClassLoader.findClass(SandboxClassLoader.java:101)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at io.appflate.restmock.RESTMockServer.init(RESTMockServer.java:54)
	at io.appflate.restmock.RESTMockServerStarter$1.run(RESTMockServerStarter.java:44)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

This exception seems to be thrown within RESTMockServer.init() method, when new MockWebServer object is trying to be created.
It's also kinda weird because everything works well in instrumented android tests.
I created a sample kotlin project with your library [here](https://github.com/bartoszwilk/RestMock) where this problem exists, so I would be very grateful if you could take a look on this in your spare time.
As you will see, I included there all stuff from your robolectric sample.

Have you ever met similar problem while working on your library?

**Comment by ntoskrnl** [comment]
We are facing the same problem. Here is a more detailed stacktrace:

```
okhttp3.mockwebserver.MockWebServer$3 execute
WARNING: MockWebServer[35852] failed unexpectedly
java.lang.NoClassDefFoundError: okhttp3/internal/framed/FramedConnection$Listener
    at okhttp3.mockwebserver.MockWebServer.serveConnection(MockWebServer.java:404)
    at okhttp3.mockwebserver.MockWebServer.access$800(MockWebServer.java:102)
    at okhttp3.mockwebserver.MockWebServer$3.acceptConnections(MockWebServer.java:379)
    at okhttp3.mockwebserver.MockWebServer$3.execute(MockWebServer.java:345)
    at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: okhttp3.internal.framed.FramedConnection$Listener
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 8 more
```

In our project we are using OkHttp 3.6.0, and this library depends on 3.4.1 – maybe  this is the problem?

**Comment by Oknesif** [comment]
I had the same problem and solved it by forcing newer mockServer dependency.

configurations.all {
    resolutionStrategy {
        force 'com.squareup.okhttp3:mockwebserver:3.6.0'
    }
}

**Comment by andrzejchm** [comment]
I will make sure to update the `okHttp` libraries to the lastest version in the next release. Meanwhile, @Oknesif's solution should do the trick

**Comment by andrzejchm** [comment]
this should be fixed with the `0.2.2` version, which now works with okhttp version `3.7.0`. Stay tuned for the release
